src/Stratis.Bitcoin.IntegrationTests/Wallet/WalletTests.cs

```diff
@@ -69,8 +69,8 @@ public void WalletCanReceiveAndSendCorrectly()
                 Assert.Null(stratisReceiver.FullNode.WalletManager().GetSpendableTransactionsInWallet("mywallet").First().Transaction.BlockHeight);
 
                 // generate two new blocks do the trx is confirmed
-                stratisSender.GenerateStratis(1, new List<Transaction>(new[] { stratisSender.FullNode.Network.CreateTransaction(trx.ToBytes()) }));
-                stratisSender.GenerateStratis(1);
+                stratisSender.GenerateBlocks(1, new List<Transaction>(new[] { stratisSender.FullNode.Network.CreateTransaction(trx.ToBytes()) }));
```

Comment:
what's the difference between the two methods?
If I want to write a test, which one should I use?

Reply:
@bokobza this is a tricky one, there are 6 tests (I think) that use this from a valid perspective. But they are from 2 test classes... One option would be to make those methods internal to the class so that other devs can't use them? We could properly comment them maybe?

Reply:
`GenerateStratisWithMiner` uses the actual mining feature `GenerateBlocks` mines using test code that's why I wanted to get rid of it.

Reply:
It's a bit strange because the two methods do almost the same thing, but in completely different ways.

Reply:
Even if we prevent the devs to use this, it's not great.
Maybe a solution would be to throw trxs in the mempool first and then call `GenerateStratisWithMiner`.
If there is nothing we can do then at least rename GenerateBlocks (maybe back to GenerateStratis) and add some comments on it.

Reply:
>   Maybe a solution would be to throw trxs in the mempool first and then call GenerateStratisWithMiner

This might be a great solution! Let me investigate 👍

Reply:
@dangershony yeah sure but these are called next to each other, it's a bit confusing.

Reply:
@fassadlr and it's more integration than constructing blocks manually. 🐱‍👓

Reply:
Ah yeah I did not realise its a rename, indeed I agree we should delete this method all together and chuck trx in mempool

Reply:
This is actually a tricky one as we need to somehow inject a fake block with double spend transactions into the puller so that consensus can reject it. If we add a tx to the mempool and try and a double spend it, it breaks (correctly) So the test's assert then needs to be changed OR we move that method that creates the double spend blocks into the class.

Reply:
Look at `TxMempoolBlockDoublespend`

Reply:
As agreed keep this method for special cases and all others use the node miner.